mod/wikirate_source/spec/set/type/source/source_spec.rb

```diff
@@ -61,182 +85,184 @@
+          expect(return_source_card.errors[:source])
+            .to include(' can only be source type or valid URL.')
         end
       end
-      context "while link is a non existing card" do
-        it "returns error " do
+      context 'while link is a non existing card' do
+        it 'returns error ' do
           Card::Env.params[:sourcebox] = 'true'
-          return_source_card = Card.new :type_id=>Card::SourceID,:subcards=>{ '+Link' => {:content=> "this is not a exisiting card"}, '+File' =>{:type_id=>Card::FileID}, '+Text'=>{:type_id=>Card::BasicID,:content=>""} }
+          return_source_card = Card.new source_args(
+            link: "this is not a exisiting card"
+          )
           expect(return_source_card).not_to be_valid
           expect(return_source_card.errors).to have_key :source
-          expect(return_source_card.errors[:source]).to include(" does not exist.")
+          expect(return_source_card.errors[:source])
+            .to include(' does not exist.')
 
         end
       end
     end
   end
-  describe "while rendering views" do
+  describe 'while rendering views' do
     before do
       login_as 'joe_user'
       @url = 'http://www.google.com/?q=wikirateissocoolandawesomeyouknow'
       @source_page = create_page @url,{}
     end
-    it "renders titled view with voting" do
+    it 'renders titled view with voting' do
       expect(@source_page.format.render_titled).to eq(@source_page.format.render_titled_with_voting)
     end
 
-    it "renders open view with :custom_source_header to be true" do
+    it 'renders open view with :custom_source_header to be true' do
       expect(@source_page.format.render_open).to include(@source_page.format.render_header_with_voting)
     end
 
-    it "renders header view with :custom_source_header to be true" do
-      expect(@source_page.format.render_header  :custom_source_header=>true ).to include(@source_page.format.render_header_with_voting)
+    it 'renders header view with :custom_source_header to be true' do
+      expect(@source_page.format.render_header  custom_source_header: true ).to include(@source_page.format.render_header_with_voting)
     end
-    it "renders metric_import_link" do
+    it 'renders metric_import_link' do
       test_csv = File.open("#{Rails.root}/mod/wikirate_source/spec/set/type_plus_right/source/import_test.csv")
-      sourcepage = Card.create! :type_id=>Card::SourceID,:subcards=>{'+File'=>{ :file=>test_csv,:type_id=>Card::FileID}}
+      sourcepage = create_source file: test_csv
       html = sourcepage.format.render_metric_import_link
-      source_file = sourcepage.fetch :trait=>:file
-      expect(html).to have_tag("a",:with=>{:href=>"/#{source_file.cardname.url_key}?view=import"},:text=>"Import to metric values")
+      source_file = sourcepage.fetch trait: :file
+      expect(html).to have_tag("a",with: {href: "/#{source_file.cardname.url_key}?view=import"},text: 'Import to metric values')
```

Comment:
Line is too long. [128/80]<br>Space missing after comma.<br>Space inside { missing.<br>Space inside } missing.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.